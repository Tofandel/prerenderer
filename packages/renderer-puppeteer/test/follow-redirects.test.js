const path = require('path')
const Prerenderer = require('@prerenderer/prerenderer')
const Renderer = require('@prerenderer/renderer-puppeteer')

const testDir = path.resolve(__dirname, '../../../tests/history')

test('changes the route property when history.pushState is used to change the page URL', async () => {
  const prerenderer = new Prerenderer({
    staticDir: testDir,
    indexPath: path.resolve(testDir, 'push-state.html'),
    renderer: new Renderer(),
  })

  await prerenderer.initialize()
  const renderedRoutes = await prerenderer.renderRoutes(['/'])
  await prerenderer.destroy()
  expect(renderedRoutes).toMatchSnapshot()
})

test('changes the route property when history.replaceState is used to change the page URL', async () => {
  const prerenderer = new Prerenderer({
    staticDir: testDir,
    indexPath: path.resolve(testDir, 'replace-state.html'),
    renderer: new Renderer(),
  })

  await prerenderer.initialize()
  const renderedRoutes = await prerenderer.renderRoutes(['/'])
  await prerenderer.destroy()
  expect(renderedRoutes).toMatchSnapshot()
})

test('maintains the correct route value when history.pushState is followed by history.back', async () => {
  const prerenderer = new Prerenderer({
    staticDir: testDir,
    indexPath: path.resolve(testDir, 'push-state-back.html'),
    renderer: new Renderer(),
  })

  await prerenderer.initialize()
  const renderedRoutes = await prerenderer.renderRoutes(['/'])
  await prerenderer.destroy()
  expect(renderedRoutes).toMatchSnapshot()
})
