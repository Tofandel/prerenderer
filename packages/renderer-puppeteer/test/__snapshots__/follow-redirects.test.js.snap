// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`changes the route property when history.pushState is used to change the page URL 1`] = `
[
  {
    "html": "<!DOCTYPE html><html><head>
  <title>Prerenderer Test: History pushState</title>
</head>
<body>
  <script>
    document.addEventListener('DOMContentLoaded', () => {
      history.pushState({}, '', '/nested/test')
      document.body.innerHTML += '<p>Render Output</p>'
    })
  </script>


<p>Render Output</p></body></html>",
    "originalRoute": "/",
    "route": "/nested/test",
  },
]
`;

exports[`changes the route property when history.replaceState is used to change the page URL 1`] = `
[
  {
    "html": "<!DOCTYPE html><html><head>
  <title>Prerenderer Test: History replaceState</title>
</head>
<body>
  <script>
    document.addEventListener('DOMContentLoaded', () => {
      history.replaceState({}, '', '/nested/test')
      document.body.innerHTML += '<p>Render Output</p>'
    })
  </script>


<p>Render Output</p></body></html>",
    "originalRoute": "/",
    "route": "/nested/test",
  },
]
`;

exports[`maintains the correct route value when history.pushState is followed by history.back 1`] = `
[
  {
    "html": "<!DOCTYPE html><html><head>
  <title>Prerenderer Test: History back</title>
</head>
<body>
  <script>
    document.addEventListener('DOMContentLoaded', () => {
      history.pushState({}, '', '/nested/test')
      history.back()
      document.body.innerHTML += '<p>Render Output</p>'
    })
  </script>


<p>Render Output</p></body></html>",
    "originalRoute": "/",
    "route": "/",
  },
]
`;
